modules/datahandling/projectrequests_test.go

```diff
@@ -1,33 +1,535 @@
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	response, notification, err2 := newRequest.process()
-	if err2 != nil {
-		t.Fatal(err2)
+	// are we notifying the right people
+	if len(closures) != 1 ||
+		reflect.TypeOf(closures[0]).String() != "datahandling.chanSubscribeClosure" {
+		t.Fatalf("did not properly process, recieved %d closure(s)", len(closures))
+	}
+
+	sub := closures[0].(chanSubscribeClosure)
+	// did the server return success status
+	val, err := strconv.ParseInt(sub.key, 10, 64)
+	if val != req.ProjectID {
+		t.Fatalf("Subscribe function wanted to subscribe to the wrong channel")
+	}
+}
+
+func TestProjectUnsubscribe_Process(t *testing.T) {
+	configSetup()
+	req := *new(projectUnsubscribeRequest)
+	setBaseFields(&req)
+	db := dbfs.NewDBMock()
+
+	req.Resource = "Project"
+	req.Method = "Unsubscribe"
+	req.ProjectID = 1
+
+	closures, err := req.process(db)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// are we notifying the right people
+	if len(closures) != 1 ||
+		reflect.TypeOf(closures[0]).String() != "datahandling.chanUnsubscribeClosure" {
+		t.Fatalf("did not properly process, recieved %d closure(s)", len(closures))
+	}
+
+	sub := closures[0].(chanUnsubscribeClosure)
+	// did the server return success status
+	val, err := strconv.ParseInt(sub.key, 10, 64)
+	if val != req.ProjectID {
+		t.Fatalf("Subscribe function wanted to subscribe to the wrong channel")
+	}
+}
+
+func TestProjectDeleteRequest_process(t *testing.T) {
+	configSetup()
+	req := *new(projectDeleteRequest)
+	setBaseFields(&req)
+
+	req.Resource = "Project"
+	req.Method = "Delete"
+
+	db := dbfs.NewDBMock()
+	usermeta := dbfs.UserMeta{
+		FirstName: "Gene",
+		LastName:  "Logan",
+		Email:     "loganga@codecollaborate.com",
+		Password:  "correct horse battery staple",
+		Username:  "loganga",
+	}
+	db.Users["loganga"] = usermeta
+	projID, err := db.MySQLProjectCreate("loganga", "new project")
+
+	db.FunctionCallCount = 0
+	req.ProjectID = projID
+
+	closures, err := req.process(db)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// didn't call extra db functions
+	if db.FunctionCallCount != 1 {
+		t.Fatal("did not call correct number of db functions")
+	}
+
+	// are we notifying the right people
+	if len(closures) != 2 ||
+		reflect.TypeOf(closures[0]).String() != "datahandling.toSenderClosure" ||
+		reflect.TypeOf(closures[1]).String() != "datahandling.toChannelClosure" {
+		t.Fatalf("did not properly process, recieved %d closure(s)", len(closures))
+	}
+
+	resp := closures[0].(toSenderClosure).msg.ServerMessage.(response)
+	not := closures[1].(toChannelClosure).msg.ServerMessage.(notification)
+	// did the server return success status
+	if resp.Status != success {
+		t.Fatalf("Process function responded with status: %d", resp.Status)
 	}
 
-	if response == nil || notification == nil {
-		// TODO: check if process worked
-		fmt.Println("unimplemented response and notification logic!")
+	// is the data actually correct
+	if not.ResourceID != projID {
```

Comment:
hmm, I get where you're coming from because it definitely shows the direction of the expectation, but honestly it seems likes it actually takes a bit more time to mentally parse